packages/openapi3/test/tagmetadata.test.ts

```diff
@@ -0,0 +1,249 @@
+  });
+});
+
+it.each([
+  [
+    "tagMetadata without additionalInfo",
+    `@tagMetadata("tagName")`,
+    { tagName: { name: "tagName" } },
+  ],
+  [
+    "tagMetadata without externalDocs",
+    `@tagMetadata("tagName",{description: "Pets operations"})`,
+    { tagName: { name: "tagName", description: "Pets operations" } },
+  ],
+  [
+    "multiple tagsMetadata",
+    `@tagMetadata(
+        "tagName1",
+        {
+          description: "Pets operations",
+          externalDocs: {
+            url: "https://example.com",
+            "x-custom": "string"
+          },          
+        }
+      )
+      @tagMetadata(
+        "tagName2",
+        {
+          description: "Pets operations",
+          externalDocs: {
+            url: "https://example.com",
+            description: "More info.",           
+          },
+          "x-custom": "string"
+        }
+      )`,
+    {
+      tagName1: {
+        name: "tagName1",
+        description: "Pets operations",
+        externalDocs: {
+          url: "https://example.com",
+          "x-custom": "string",
+        },
+      },
+
+      tagName2: {
+        name: "tagName2",
+        description: "Pets operations",
+        externalDocs: {
+          url: "https://example.com",
+          description: "More info.",
+        },
+        "x-custom": "string",
+      },
+    },
+  ],
+])("%s", async (_, code, expected) => {
+  const runner = await createOpenAPITestRunner();
+  const { PetStore } = await runner.compile(
+    `
+    ${code}
+    @test 
+    namespace PetStore {}
+    `,
+  );
+  deepStrictEqual(getTagsMetadata(runner.program, PetStore), expected);
+});
+
+it.each([
+  [
+    "set the additional information with @tagMetadata decorator",
+    `@tag("TagName") op NamespaceOperation(): string;`,
+    [
+      {
+        name: "TagName",
+        description: "Pets operations",
+        externalDocs: {
+          description: "More info.",
+          url: "https://example.com",
+          "x-custom": "string",
+        },
+        "x-custom": "string",
+      },
+    ],
+  ],
+  [
+    "set tag with @tagMetadata decorator",
+    ``,
+    [
+      {
+        name: "TagName",
+        description: "Pets operations",
+        externalDocs: {
+          description: "More info.",
+          url: "https://example.com",
+          "x-custom": "string",
+        },
+        "x-custom": "string",
+      },
+    ],
+  ],
+  [
+    "set tags with @tagMetadata decorator and @tag decorator",
+    `@tag("opTag") op NamespaceOperation(): string;`,
+    [
+      { name: "opTag" },
+      {
+        name: "TagName",
+        description: "Pets operations",
+        externalDocs: {
+          description: "More info.",
+          url: "https://example.com",
+          "x-custom": "string",
+        },
+        "x-custom": "string",
+      },
+    ],
+  ],
+])("%s", async (_, code, expected) => {
+  const res = await openApiFor(
```

Comment:
resue the following? 
and it.each([
[`test description...`, `decorator....`, `operationDeclaration...`, [{expected diag},{...}] | undefined, {expected output} | undefined]
] ...)
```
    @service
    ${tagMetaDecorator}  
    namespace PetStore{${operationDeclaration}};
    
```

Reply:
you can also add a funtion `createTestCase(testDescription: string, codeBlockWithTagMetadata: string, ...)`